templates/workshopparticipants.html

```diff
@@ -13,57 +14,61 @@ <h1>{{ workshop.title }} &dash; uczestnicy</h1>
               {{ workshop_participant.participant.user.email }}
             </td>
-            <td>
+            <td style="width: 12.5%;">
               {% if has_perm_to_edit %}
-                <input type=number step=0.1
-                       value="{{ workshop_participant.qualification_result | default_if_none:'' }}"
-                       class="points-input"
-                       data-id="{{ workshop_participant.id }}">
-                <a href="#" class="save">Zapisz</a>
+                {{ workshop_participant.form.qualification_result }}
               {% else %}
                 {{ workshop_participant.qualification_result | default_if_none:'' }}
               {% endif %}
             </td>
-            <td>
+            <td style="width: 1px; white-space: nowrap;">
+              {% if has_perm_to_edit %}
+                <button class="btn btn-default" style="padding: 6px 12px;" data-save="#{{ workshop_participant.form.qualification_result.id_for_label }}" data-participant-id="{{ workshop_participant.id }}"><span class="glyphicon glyphicon-floppy-disk"></span></button>
+              {% endif %}
+            </td>
+            <td style="width: 20%;">
               {% if has_perm_to_edit %}
-                <input type=text
-                       value="{{ workshop_participant.comment | default_if_none:'' }}"
-                       class="comment-input"
-                       data-id="{{ workshop_participant.id }}">
-                <a href="#" class="savec">Zapisz</a>
+                {{ workshop_participant.form.comment }}
               {% else %}
                 {{ workshop_participant.comment | default_if_none:'' }}
               {% endif %}
             </td>
-            <td class="qualified-mark">
-              {% load wwwtags %}
+            <td style="width: 1px; white-space: nowrap;">
+              {% if has_perm_to_edit %}
+                <button class="btn btn-default" style="padding: 6px 12px;" data-save="#{{ workshop_participant.form.comment.id_for_label }}" data-participant-id="{{ workshop_participant.id }}"><span class="glyphicon glyphicon-floppy-disk"></span></button>
+              {% endif %}
+            </td>
+            <td class="qualified-mark" style="vertical-align: middle;">
               {{ workshop_participant.is_qualified | qualified_mark }}
             </td>
           </tr>
         {% endfor %}
         </tbody>
       </table>
       <div class="containter-fluid">
-        Maile: <input type="text"
-                      value="{% for workshop_participant in workshop_participants %}{{ workshop_participant.participant.user.email }} {% endfor %}"
-                      id="participants_emails">
+        <div class="form-group">
+          <label for="participants_emails">Maile:</label>
+          <input id="participants_emails" class="form-control" value="{% for workshop_participant in workshop_participants %}{{ workshop_participant.participant.user.email }}, {% endfor %}" />
```

Comment:
@grzegorz225 This is email so in no way can this be exploited, but I wonder if we see here a possibility of exploits for some Django apps (if you control a html tag parameter can you do something like: `" onclick="alert(1);`?)

Reply:
You can't, unless you specifically tell Django to disable autoescaping

https://docs.djangoproject.com/en/3.0/ref/templates/language/#automatic-html-escaping

Reply:
You are right `"` is escaped.